src/core/biblio.js

```diff
@@ -140,22 +140,34 @@ function bibref(conf) {
       </section>`;
     addId(sec);
 
-    refs.sort((a, b) =>
-      a.toLocaleLowerCase().localeCompare(b.toLocaleLowerCase())
-    );
-    const refObjects = refs.map(getRefContent);
+    const refObjects = { good: [], bad: [] };
+    refs.map(getRefContent).reduce((refObjects, ref) => {
+      const refType = ref.refcontent ? "good" : "bad";
+      refObjects[refType].push(ref);
+      return refObjects;
+    }, refObjects);
 
-    const aliases = refObjects.reduce((aliases, { key, ref }) => {
+    const aliases = refObjects.good.reduce((aliases, ref) => {
+      const key = ref.refcontent.id;
       const keys = !aliases.has(key)
         ? aliases.set(key, []).get(key)
         : aliases.get(key);
-      if (!keys.includes(ref.toLowerCase())) keys.push(ref);
+      keys.push(ref.ref);
       return aliases;
     }, new Map());
-    const refsToAdd = refObjects.reduce((refsToAdd, ref) => {
-      if (!refsToAdd.find(r => r.key === ref.key)) refsToAdd.push(ref);
-      return refsToAdd;
-    }, []);
+
+    const uniqueRefs = refObjects.good.reduce((uniqueRefs, ref) => {
```

Comment:
After this, `uniqueRefs` is only ever used as a an array (by spreading the values), so I wonder if you should just convert it to an array here.